src/System.Private.CoreLib/shared/System/String.Manipulation.cs

```diff
@@ -522,10 +522,12 @@ private static string FormatHelper(IFormatProvider? provider, string format, Par
             if (format == null)
                 throw new ArgumentNullException(nameof(format));
 
-            return StringBuilderCache.GetStringAndRelease(
-                StringBuilderCache
-                    .Acquire(format.Length + args.Length * 8)
-                    .AppendFormatHelper(provider, format, args));
+            ValueStringBuilder sb = new ValueStringBuilder(format.Length + args.Length * 8);
```

Comment:
A more valuable optimization here would be a fixed size stack allocated buffer, similar to other places where ValueStringBuilder is used.

Reply:
can still keep this optimization as `sb.EnsureCapacity(format.Length + args.Length * 8)`

Reply:
Done